### DISCUSSION TITLE
Маркеры крашат карту

### DISCUSSION THREAD

**Discussion by vermorag** [yandex-maps-unofficial/vue-yandex-maps#280] [Q&A (2.x)] [answered]
```
<YandexMap :settings="{ location }">
           <YandexMapMarker
                v-if="mapStorage.exactPoint !== null && coreStorage.mode === DeliveryMode.DOOR"
                :settings="{ coordinates: toRaw(mapStorage.exactPoint) }"
            >
                <DoorIcon class=":uno: cursor-pointer max-w-none transform-office-icon" />
            </YandexMapMarker>
            <YandexMapMarker
                v-if="mapStorage.geolocation !== null"
                :settings="{ coordinates: toRaw(mapStorage.geolocation) }"
            >
                <div
                    class=":uno: w-3.5 h-3.5 rounded-full border-2 border-brand-peak-main bg-brand-tertiary-main shadow transform-point-icon"
                />
            </YandexMapMarker>
        </YandexMap>
```

Если сначала через v-if будет показан нижний маркер, а потом сработает v-if для верхнего, то будет вот такая ошибка:
```
DOMException: Node.insertBefore: Child to insert before is not a child of this node
```
![image](https://github.com/yandex-maps-unofficial/vue-yandex-maps/assets/25276845/a331ff0e-2b1c-4c28-93e2-bc4876b7dc1a)

А если в обратном порядке (сначала верхний по DOM-у, потом нижний), то такой ошибки нет и все рендерится нормально.

Проходился дебаггером - он зачем-то пытается маркер вставить в ноду `__ymap_slots` рядом с уже существующим `__ymap-marker` и там его не находит, тк маркеры все не в слотах, а на карте.

Куда копать дальше не знаю. Есть идеи?

**Comment by daniluk4000** [comment]
Привет! Создай плиз ишью. 

Я с этим недавно сталкивался, единственный способ это исправить - накидать key на маркеры. 

Идеи есть, поищу способ решения, пока что ключ как костыль. То, что решить получится (без потери перфоманса), я не уверен, но я сделаю всё возможное) Скорее всего, ошибка вызвана моими оптимизациями (в том числе), и варианты будут такие:
- Я решу https://github.com/yandex-maps-unofficial/vue-yandex-maps/issues/276 и это больше не потребуется
- Я удалю оптимизацию для фикса в жертву перфоманса
- Исправить не получится и придется работать с key

**Reply by vermorag** [reply]
Создал #281, но гитхаб не дает на нее лейблы накинуть - прав не хватает

**Reply by daniluk4000** [reply]
Ща бахну. Спасибо!

**Reply by vermorag** [reply]
Попробовал :key накиунть - тоже самое.
![image](https://github.com/yandex-maps-unofficial/vue-yandex-maps/assets/25276845/34ea8020-d458-4b61-97e5-0e9b9553c681)

**Reply by daniluk4000** [reply]
Воспроизвел локально. Сейчас сам подебажу

**Reply by daniluk4000** [reply, marked as answer]
Дайте мне эти выходные и я это решу. Без потери перфоманса. Решение я уже нашел. 

Пока что сделайте странную вещь))

Просто оберните оба маркера в `<template>`. Ключи можете убрать

**Reply by daniluk4000** [reply]
Напишите потом пжл, поможет или нет. Мне помогло в базовом примере

**Reply by vermorag** [reply]
На удивление - работает.

**Reply by daniluk4000** [reply]
Я тоже удивился. Сначала я хотел предложить в див обернуть, но тут вон даже template хватило.

В выходные буду изучать этот вопрос подробнее. Я уже написал фикс, но он ломает обратную совместимость, мне там придется немного посидеть

**Reply by vermorag** [reply]
Понял. Спасибо большое.
